index.js

```diff
@@ -10,8 +10,51 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+// retrieving colors data using fetch:
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  // fetch data from API:
+  const response = await fetch(COLORS);
+
+  if(!response.ok) {
+    throw Error('Error: Looks like something went wrong. Failed to fetch colors.');
+  }
+
+  const colors = await response.json();
+
+  let filteredColors = colors;
+
+  // filter by name
+  if(name) {
```

Comment:
If you use a single `filter` call, you can move all the filtering into that. This way if down the line we need to combine filters you can do so.